src/client/render2d/pixi_renderer.tsx

```diff
@@ -0,0 +1,119 @@
+import * as classNames from 'classnames'
+import { IReactionDisposer } from 'mobx'
+import { observable } from 'mobx'
+import { action } from 'mobx'
+import { autorun } from 'mobx'
+import { observer } from 'mobx-react'
+import { WebGLRenderer } from 'pixi.js'
+import { CanvasRenderer } from 'pixi.js'
+import { Container } from 'pixi.js'
+import { autoDetectRenderer } from 'pixi.js'
+import * as React from 'react'
+import { Component } from 'react'
+import ReactResizeDetector from 'react-resize-detector'
+
+import { Transform } from '../math/transform'
+
+import { renderObject2d } from './pixi/rendering'
+import { RendererProps } from './renderer_props'
+import * as style from './style.css'
+
+@observer
+export class PixiRenderer extends Component<RendererProps> {
+  @observable private resolution: Transform = Transform.of()
+  private canvas: HTMLCanvasElement
+  private renderer: WebGLRenderer | CanvasRenderer
+  private stopAutorun: IReactionDisposer
+
+  componentDidMount() {
+    if (!this.canvas) {
+      return
+    }
+
+    this.renderer = autoDetectRenderer({
+      view: this.canvas,
+      transparent: true,
+      antialias: true,
+    })
+
+    // Render when changes happen
+    this.stopAutorun = autorun(() => this.renderCanvas(), {
+      scheduler: requestAnimationFrame,
+    })
+  }
+
+  componentWillUnmount() {
+    if (this.stopAutorun) {
+      this.stopAutorun()
+    }
+  }
+
+  render() {
+    return (
+      <div className={classNames(this.props.className, style.container)}>
+        <ReactResizeDetector handleWidth handleHeight onResize={this.onResize} />
+        <canvas
+          className={style.container}
+          width={-this.resolution.translate.x * 2}
+          height={-this.resolution.translate.y * 2}
+          ref={this.onRef}
+        />
+      </div>
+    )
+  }
+
+  private onRef = (canvas: HTMLCanvasElement) => {
+    this.canvas = canvas
+  }
+
+  renderCanvas = () => {
+    // Render our scene
+    const { scene, camera } = this.props
+
+    const transform = this.resolution.inverse().then(camera)
+
+    const cam = new Container()
+    cam.rotation = transform.rotate
+    cam.x = transform.translate.x
+    cam.y = transform.translate.y
+    cam.scale.x = transform.scale.x
+    cam.scale.y = transform.scale.y
+    cam.addChild(renderObject2d(this.props.scene))
+
+    this.renderer.render(cam)
```

Comment:
I'm confused what's going on here, does the `renderObject2d` call above do the rendering? or does this line?

Reply:
renderObject2d is the transformer that transforms the scene graph into a pixi scene graph. The function was taken from canvas and probably would have a better name in this case

Reply:
Right, yeah in normal 2d canvas land side-effects happen along the way, and hence `render` is a good prefix. It looks like pixi takes a scene graph similar to three.js and the side-effects happen later, so it doesn't actually do rendering here.

I call this the view model in three.js land. Perhaps here it should be named something like `toPixiScene`?